JASP-Engine/JASP/R/descriptives.R

```diff
@@ -1427,4 +1455,89 @@ Descriptives <- function(jaspResults, dataset, options) {
     return(TRUE)
   else
     return(FALSE)
-}
\ No newline at end of file
+}
+
+.Xbarchart <- function(jaspResults, dataset, options, variables){
+
+  ready <- (length(options$variables) > 0)
+
+  if (ready && is.null(jaspResults[["Xbarchart"]]))
+    .XbarPlotDescriptives(jaspResults, dataset, options, ready)
+
+  return()
+}
+
+.XbarPlotDescriptives <- function(jaspResults, dataset, options, ready){
+
+  XbarPlot <- createJaspPlot(title = "X bar chart", width=500, aspectRatio=1)
+  jaspResults[["Xbarchart"]] <- XbarPlot
+
+  if (!ready)
+    return()
+
+  .XbarFillPlotDescriptives(XbarPlot, dataset, options)
+
+  return()
+}
+
+.XbarFillPlotDescriptives <- function(XbarPlot, dataset, options){
+
+  plot <- qcc::qcc(dataset, type="xbar", plot= F)
+
+  if (inherits(plot, "try-error")){
+    errorMessage <- as.character(plot)
+    XbarPlot$setError(errorMessage)
+    return()
+  }
+
+  XbarPlot$plotObject <- plot
+
+  return()
+}
+
+.Rchart <- function(jaspResults, dataset, options, variables){
+
+  ready <- (length(options$variables) > 0)
+
+  if (ready && is.null(jaspResults[["Rchart"]]))
+    .RchartPlotDescriptives(jaspResults, dataset, options, ready)
+
+  return()
+}
+
+
+
+## plot function
+
+.RchartPlotDescriptives <- function(jaspResults, dataset, options, ready){
+
+  RchartPlot <- createJaspPlot(title = "R chart", width=500, aspectRatio=1)
+  jaspResults[["Rchart"]] <- RchartPlot
+
+  if (!ready)
+    return()
+
+
+  .RchartFillPlotDescriptives(RchartPlot, dataset, options)
+
+  return()
+}
```

Comment:
These functions can also be merged like the ones above.  Also, the empty whitespace and comments here are redundant. If you have a function title "xxxFillPlot" then a comment like "Filling function" doesn't add anything.